documentation/demo.js

```diff
@@ -0,0 +1,79 @@
+/**
+ * To run this example, you must
+ * 1. git checkout https://github.com/daostack/client
+ * 2. npmm install
+ * 3. npm run build // build th epackage
+ * 3. docker-compose up -d // this will start docker containers with the test services
+ * 4. node documentation/demo.js // run this file
+ * 5. docker-compose down // stop the docker containers when you are done
+ */
+
+async function main() {
+
+  console.log('hello!')
+
+  // mport from the local build of the library (that was created with npm run build)
+  // typically, one would use require('@daostack/client')
+  const { Arc } = require('../dist/lib/index.js')
+
+  // "Arc" is the main class that handles configuration and connections to various services
+  // create an Arc instanc with settings to connect to the local docker images
+  const arc = new Arc({
+    graphqlHttpProvider: 'http://127.0.0.1:8000/subgraphs/name/daostack',
+    graphqlHttpMetaProvider: 'http://127.0.0.1:8000/subgraphs',
+    graphqlWsProvider: 'http://127.0.0.1:8001/subgraphs/name/daostack',
+    web3Provider: 'ws://127.0.0.1:8545',
+    ipfsProvider: '/ip4/127.0.0.1/tcp/5001',
+  })
+  // we must provice Arc with some contract information.
+  // We could have hardcoded it, but get this information from the subgraph
+  const contractInfos = await arc.fetchContractInfos()
+  arc.setContractInfos(contractInfos)
+
+  // we get the first returned item from the obervable that returns a list of DAOs
+  const daos = await arc.daos().first()
+
+  console.log(`Found ${daos.length} DAOs in ${arc.graphqlHttpProvider}`)
+
+  // given the ida of a DAO, we can also create a resh DAO instance
+  const { DAO } = require('../dist/lib/index.js')
+  const dao = new DAO(daos[0].id, arc)
+
+  // get the DAO state (again, the "first()" object from the observable)
+  const daoState  = await dao.state().first()
+  // the state contains information such as the address, name or nativeToken of the DAO
+  console.log(`This DAO has name "${daoState.name}" and is deployed on ${daoState.address}`)
+
+  // to create a proposal, we must first find the address of the Scheme to create it in
+  const schemes = await dao.schemes({ where: { name: 'ContributionReward'}}).first()
+
+  if (schemes.length === 0) {
+    throw Error('Something went wrong - no ContrsbutsonReward scheme was registered with this DAO')
+  }
+  const schemeState = await schemes[0].state().first()
+
+  console.log(`We'll create a ${schemeState.name} proposal at the scheme at ${schemeState.address}`)
+
+  // first construct a transaction
+  const tx = dao.createProposal({
```

Comment:
Would be great if this `createProposal` function was overloaded to either take in the scheme's address or a type Enum of the scheme. If multiple schemes of that type are found, an error could be thrown.

I could understand why this wouldn't be desired though, because of the error case described above.

Reply:
Yes, it probably would be nicer if you woulc just "create a new contribution reward proposal" without bothering to find out what the scheme adddress is.

not sure about this, though, not in this PR i'd say :-)